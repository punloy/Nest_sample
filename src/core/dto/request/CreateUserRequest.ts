import { ApiProperty } from "@nestjs/swagger";
import { IsDefined, IsString } from "class-validator";

export class CreateUserRequest {
    @ApiProperty({ description: "Add username" })
    @IsDefined()
    @IsString()
    name: string;

    @ApiProperty({ description: "Add password" })
    @IsDefined()
    @IsString()
    password: string;
}
