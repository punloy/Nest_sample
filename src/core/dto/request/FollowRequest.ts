import { ApiProperty } from "@nestjs/swagger";
import { IsDefined, IsString } from "class-validator";

export class FollowRequest {
    @ApiProperty({ description: "Follow username" })
    @IsDefined()
    @IsString()
    name: string;
}
