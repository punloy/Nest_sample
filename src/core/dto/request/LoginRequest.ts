import { ApiProperty } from "@nestjs/swagger";
import { IsDefined, IsString } from "class-validator";

export class LoginRequest {
    @ApiProperty({ description: "Login username" })
    @IsDefined()
    @IsString()
    name: string;

    @ApiProperty({ description: "Login password" })
    @IsDefined()
    @IsString()
    password: string;
}
