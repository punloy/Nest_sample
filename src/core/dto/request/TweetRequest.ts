import { ApiProperty } from "@nestjs/swagger";
import { IsDefined, IsString } from "class-validator";

export class TweetRequest {
    @ApiProperty({ description: "Tweet tittle" })
    @IsDefined()
    @IsString()
    tittle: string

    @ApiProperty({ description: "Tweet content" })
    @IsDefined()
    @IsString()
    content: string
}