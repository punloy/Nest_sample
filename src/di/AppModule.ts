import { Module, } from '@nestjs/common';
import { HealthModule } from './HealthModule';

@Module({
  imports: [HealthModule]
})
export class AppModule { }
