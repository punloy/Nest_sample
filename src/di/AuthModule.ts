import { Module } from '@nestjs/common';
import { PassportModule } from '@nestjs/passport';
import { TypeOrmModule } from '@nestjs/typeorm';
import { JwtStrategy } from '../core/jwt/JwtStrategy';
import { LocalUsertrategy } from '../core/jwt/LocalUserStrategy';
import { tweet } from '../core/index';
import { AuthService } from '../service/AuthService';

@Module({
    imports: [
        PassportModule,
        TypeOrmModule.forFeature(Object.values(tweet))
    ],
    providers: [
        AuthService,
        LocalUsertrategy,
        JwtStrategy,
    ]
})
export class AuthModule { }