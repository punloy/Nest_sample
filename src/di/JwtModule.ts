import { Module } from "@nestjs/common";
import { JwtService } from "../service/JwtService";

@Module({
    providers: [JwtService],
    exports: [JwtService]
})
export class JwtModule { }
