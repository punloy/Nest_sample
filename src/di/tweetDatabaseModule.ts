import { DynamicModule } from "@nestjs/common";
import { TypeOrmModule, TypeOrmModuleOptions } from "@nestjs/typeorm";
import * as entities from "../core/tweet";
import config from "config";

const defaultOptions: TypeOrmModuleOptions = {
    type: "postgres",
    host: config.get("tweetDB.host"),
    port: 5432,
    username: config.get("tweetDB.username"),
    password: config.get("tweetDB.password"),
    database: config.get("tweetDB.database"),
    synchronize: config.get("tweetDB.synchronize"),
    logging: false,
    keepConnectionAlive: true,
    entities: Object.values(entities)
};

export class TweetDatabaseModule {
    public static register(options: Partial<TypeOrmModuleOptions>): DynamicModule {
        return TypeOrmModule.forRoot(Object.assign({}, defaultOptions, options));
    }
}