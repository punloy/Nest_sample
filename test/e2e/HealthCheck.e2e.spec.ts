import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import request from 'supertest';
import { AppModule } from '../../src/di/AppModule';

describe('HealthController (e2e)', () => {
    let app: INestApplication;

    beforeEach(async () => {
        const moduleFixture: TestingModule = await Test.createTestingModule({
            imports: [AppModule],
        }).compile();

        app = moduleFixture.createNestApplication();
        await app.init();
    });

    it("/health (GET)", async () => {
        const result = await request(app.getHttpServer())
            .get('/health')
            .expect(200);
        expect(JSON.parse(result.text)).toMatchSnapshot();
    });
});