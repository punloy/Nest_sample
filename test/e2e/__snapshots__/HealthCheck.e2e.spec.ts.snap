// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthController (e2e) /health (GET) 1`] = `
Object {
  "details": Object {},
  "error": Object {},
  "info": Object {},
  "status": "ok",
}
`;
